UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -551,6 +560,16 @@ public void InitializeSensors()
             m_VectorSensorBuffer = new float[numFloatObservations];
         }
 
+        void InitializeRewardProvider()
+        {
+            // Look for a legacy reward provider.
+            m_RewardProviderComponent = GetComponent<LowLevelRewardProviderComponent>();
```

Comment:
Feels like we should be looking for an abstract class here, e.g. RewardProviderComponent, and make LowLevelRewardProviderComponent inherit from RewardProviderComponent. Otherwise what are users supposed to implement if they want to make their own reward provider?

Reply:
Yeah, I had one previously.  I'll add it back.